### ISSUE TITLE
Add optional HTTP post options

### ISSUE THREAD

**Issue by micwiec** [johnsusek/praeco#267]
@johnsusek 
Can you add in HTTP POST  UI option to define `http_post_static_payload`?

In ElastAlert YAML it might like this:
```yml
[...]
http_post_static_payload:
    error_id: XYZ_123
    error_description: "There is a problem with XYZ."
```

ElastAlert Doc: [https://elastalert.readthedocs.io/en/latest/ruletypes.html#http-post](https://elastalert.readthedocs.io/en/latest/ruletypes.html#http-post)

**Comment by nsano-rururu** [comment]
If you can develop it, please make a pull request.

**Comment by nsano-rururu** [comment]
error_id and error_description are different item names for each app, aren't they? Hmmm. It may be difficult with the current construction.

**Comment by nsano-rururu** [comment]
The combination of key and value must be variable.

**Comment by nsano-rururu** [comment]
Currently, there are no plans to support it. The reason is that I don't know how to implement it.

**Comment by nsano-rururu** [comment]
Will not be supported

**Comment by dominikstohl** [comment]
hello everyone,
does the http_post_static_payload work with the elastalert-server by praeco?
I got the following error:

`
11:16:23.964Z ERROR elastalert-server:

    ProcessController:  ERROR:elastalert:Traceback (most recent call last):

      File "/opt/elastalert/elastalert/elastalert.py", line 1537, in alert

        return self.send_alert(matches, rule, alert_time=alert_time, retried=retried)

      File "/opt/elastalert/elastalert/elastalert.py", line 1631, in send_alert

        alert.alert(matches)

      File "/opt/elastalert/elastalert/alerters/httppost.py", line 33, in alert

        payload.update(self.post_static_payload)

    ValueError: dictionary update sequence element #0 has length 1; 2 is required
`

**Comment by nsano-rururu** [comment]
@dominikstohl 

I have deleted the comment because it has nothing to do with this request.

**Comment by nsano-rururu** [comment]
@dominikstohl

The attitude of not trying to read the error message is bad. It is not a problem with elastalert-server and praeco as it is an error occurring in elastalert2.

**Comment by dominikstohl** [comment]
Hey @nsano-rururu , thanks for your answer.
I use the elastalert-server provided by praeco, so I just want to make sure thats not a problem with it.
Just to understand, elastalert-server by praeco uses elastalert2 as it is under the hood?

I try to find a solution for this problem, thank you for your fast reply and time.

**Comment by nsano-rururu** [comment]
@dominikstohl 

I agree with that understanding. Rule execution is elastalert2.
https://user-images.githubusercontent.com/611996/52892144-90a19300-3155-11e9-8050-cb4a440411a4.png

elastalert-server is a webapi created with node.js. praeco is the front end. To get the information to be displayed on the screen, elastalert-server directly accesses the index of elastalert2 of elasticsearch or executes the command of elastalert2. When running with docker, elastalert2 runs inside the container of elastalert-server and executes the set rules.
In other words, even if you specify localhost in the container, it will be in the docker container instead of the docker host. As for the address, you have to specify the IP address (global) or domain that can be accessed from inside the container to the outside.

**Comment by nsano-rururu** [comment]
We always welcome pull requests.

**Comment by Sinistr0** [comment]
Any predictions when the http_post_ static payload option will be added?

**Comment by nsano-rururu** [comment]
No plans to support
